vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -336,6 +336,74 @@ func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isRead
 	return mountpoint, fs.MountWithID(mountpoint, fstype, id, isReadOnly)
 }
 
+// private function that does the job of mounting volume in conjunction with refcounting
+// Called under a lock if refcounting hasn't been successful, else without a lock
+// Don't do refcounting, just the mount if refcounting hasn't been successful.
+func (d *VolumeDriver) processMount(r volume.MountRequest, doRefcnt bool) volume.Response {
+	// If the volume is already mounted , just increase the refcount.
+	//
+	// Note: We are deliberately incrementing refcount first, before trying
+	// to do anything else. If Mount fails, Docker will send Unmount request,
```

Comment:
yeah docker doesn't send unmount if mount fails. This comment isn't valid anymore.